app/questionnaire/path_finder.py

```diff
@@ -303,21 +302,22 @@ def _remove_current_blocks_answers_for_backwards_routing(
     def _remove_current_blocks_answers_for_new_backwards_routing(
         self, rules: dict, answer_ids_for_current_block: list[str]
     ) -> None:
-        operands = self.schema.get_operands(rules)
-        for rule in operands:
-            if isinstance(rule, dict) and (
-                "identifier" in rule
-                and rule["identifier"] in answer_ids_for_current_block
-            ):
-                if (
+        if isinstance(rules, Mapping) and QuestionnaireSchema.has_operator(rules):
```

Comment:
How come this is at this level?. It doesn't match what we had before.

Reply:
Removed this now